common/common.go

```diff
@@ -433,28 +434,74 @@ func InArray(val, array interface{}) (exists bool, index int) {
 	return
 }
 
-// Errors defines multiple errors
-type Errors []error
+// AppendError appends error in a more idiomatic way. This can start out as a
+// standard error e.g. err := errors.New("random error")
+// err = AppendError(err, errors.New("another random error"))
+func AppendError(original, incoming error) error {
+	if incoming == nil {
+		if _, file, no, ok := runtime.Caller(1); ok {
+			log.Errorf(log.Global, "AppendError() failed: incoming error is nil %s#%d\n", file, no)
```

Comment:
As fun as this is, this causes friction for say, appending an error where you don't yet know it something like 
`err = gctcommon.AppendError(err, okx.SendHTTPRequest(wow))`
Investigating the caller like this seems very debug thing to do. Are you doing this to dictate that one should not use this function like the snippet above? Even then, it seems somewhat unnecessary to call out and could be skipped and I'd rather you shame me in PR comments

For the sake of testing, I ran a benchmark:
```
BenchmarkButts-10    	 3082971	       387.7 ns/op	     288 B/op	       5 allocs/op
```

Without this `runtime` usage:
```
BenchmarkButts-10    	566423924	         2.035 ns/op	       0 B/op	       0 allocs/op
```

Reply:
Well the design of it was supposed to be like var errs error and then you check the err for not nil then append if necessary. What you are doing above cleans up a bit of code though. So I am happy to just skip nils, I just wanted to be quite strict and alert and log if it was nil for things like if `err == nil { }` which was done by accident.